packages/cli/src/metadataGeneration/typeResolver.ts

```diff
@@ -420,6 +420,20 @@ export class TypeResolver {
         return stringMetaType;
       }
 
+      if (typeReference.typeName.text === 'Record' && typeReference.typeArguments?.length === 2) {
+        const indexType = new TypeResolver(typeReference.typeArguments[0], this.current, this.parentNode, this.context).resolve();
+        if (indexType.dataType === 'string') {
```

Comment:
Why only for string? We should be able to handle numbers, boolean, string enums, number enums etc.?

Reply:
thanks for the number callout! I've added support here for numbers and test cases to verify

- number enums and string enums are handled separately (I'm not sure where, but as you can see from the test suite they are handled correctly). They actually produce fairly different OpenAPI results, because of the way swagger handles dictionaries
- `boolean` is not a valid `Record` key type; the type for `Record`'s key is `string | number | symbol`. `symbol` isn't supported by tsoa it seems (which makes sense, because it doesn't belong in a public-facing API)